From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm: correct arg in reclaim_pages()/reclaim_clean_pages_from_list()
Date: Mon, 17 Apr 2023 19:48:07 +0800

Both of them change the arg from page_list to folio_list when convert them
to use a folio, but not the declaration, let's correct it, also move the
reclaim_pages() from swap.h to internal.h as it only used in mm.

Link: https://lkml.kernel.org/r/20230417114807.186786-1-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Reviwed-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Reviewed-by: SeongJae Park <sj@kernel.org>
Reviewed-by: David Hildenbrand <david@redhat.com>
Cc: Hugh Dickins <hughd@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/swap.h |    1 -
 mm/internal.h        |    3 ++-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/include/linux/swap.h~mm-correct-arg-in-reclaim_pages-reclaim_clean_pages_from_list
+++ a/include/linux/swap.h
@@ -442,7 +442,6 @@ extern unsigned long shrink_all_memory(u
 extern int vm_swappiness;
 long remove_mapping(struct address_space *mapping, struct folio *folio);
 
-extern unsigned long reclaim_pages(struct list_head *page_list);
 #ifdef CONFIG_NUMA
 extern int node_reclaim_mode;
 extern int sysctl_min_unmapped_ratio;
--- a/mm/internal.h~mm-correct-arg-in-reclaim_pages-reclaim_clean_pages_from_list
+++ a/mm/internal.h
@@ -804,8 +804,9 @@ extern unsigned long  __must_check vm_mm
         unsigned long, unsigned long);
 
 extern void set_pageblock_order(void);
+unsigned long reclaim_pages(struct list_head *folio_list);
 unsigned int reclaim_clean_pages_from_list(struct zone *zone,
-					    struct list_head *page_list);
+					    struct list_head *folio_list);
 /* The ALLOC_WMARK bits are used as an index to zone->watermark */
 #define ALLOC_WMARK_MIN		WMARK_MIN
 #define ALLOC_WMARK_LOW		WMARK_LOW
_
